Replace pre-release CHANGELOG entries with consolidated 6.2.0 entry (#7107)
alphagov/govuk-frontend: CHANGELOG.md

@use "pkg:govuk-frontend/components/button";
```

> If you use `@import` with `pkg:` URLs, there is a known issue with how [Sass loads `pkg:` URLs with `@import`](https://github.com/sass/sass/issues/4224). We recommend only using `pkg:` URLS with `@use` or explicitly including `index.import` in the URLs you import, for example: `@import "pkg:govuk-frontend/index.import"` or `@import "pkg:govuk-frontend/components/button/index.import"`.

Author: @calvin-lau-sig7 Went with @NickColley's more detailed explanation to give teams more guidance. Happy to make any content edit before merging or go with the simpler:

> Due to how [Sass will currently load `pkg:` URLs with `@import`](https://github.com/sass/sass/issues/4224), you should only use `pkg:` URLs with `@use`.

@use "pkg:govuk-frontend/components/button";
```

> > If you're using `@import` with `pkg:` URLs there's a known issue with how [Sass loads `pkg:` URLs with `@import`](https://github.com/sass/sass/issues/4224).

Reviewer: I'm not gonna block this going out because of this but something to think about in future: I don't think using quote syntax is a good way to highlight important content _in github_. Github's design for quote syntax, especially on UI's using dark mode, actually visually de-emphesises this content:

<img width="1024" height="194" alt="Image" src="https://github.com/user-attachments/assets/43b12120-fa8c-4570-908f-311b73112f22" />

Author: GitHub supports [Alerts for giving blocks of content some emphasis and meaning](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts),  if we want something a little shinier in the CHANGELOG and Release notes. However it's not standard markdown if it ever gets rendered by something else than GitHub.

Reviewer 2: We're using them for pre-releases for the warnings, I think we either dont quote it (indent) or we use one of the shiny GitHub alerts.

>
> We recommend you either:
>

> > - only use `pkg:` URLs with `@use

Reviewer: Missed closing backtick

```suggestion
> - only use `pkg:` URLs with `@use`
```

> - make sure to include `index.import` in the URLs you import (for example: `@import "pkg:govuk-frontend/index.import"`)
>
> You should migrate any existing uses of `@import` with `pkg:` in your service.

We made this change in [pull request #6861: Resolve `pkg:` URLs from `dist/govuk` and update the review app](https://github.com/alphagov/govuk-frontend/pull/6861).

#### Add custom classes and attributes to the File upload component's wrapper

We've introduced two new parameters to the File upload component's Nunjucks macro: `wrapperClasses` and `wrapperAttributes`.

These parameters allow you to define custom classes and HTML attributes for the wrapper of the improved version of the File upload component.

```njk
{{ govukFileUpload({
  javascript: true,
  wrapperClasses: "my-custom-class",
  wrapperAttributes: {
    "data-attribute": "value"
  }
}) }}
```

We made this change in [pull request #6933: Code improvements to File upload component](https://github.com/alphagov/govuk-frontend/pull/6933).

### Recommended changes

#### Rename the `govuk-drop-zone` class on the improved File upload component

> We've changed the class name of the element that wraps the improved File upload component from `govuk-drop-zone` to `govuk-file-upload-wrapper`. The class name now better describes the function the element plays in the component.

Reviewer 3: @seaemsi and I thought this would help make this sentence easier to read.

```suggestion
We've changed the class name of the element that wraps the improved File upload component from `govuk-drop-zone` to `govuk-file-upload-wrapper`. `govuk-file-upload-wrapper` better describes what the element does.
```
